### ISSUE TITLE
pip install streamlit pandas numpy plotly prophet

### ISSUE THREAD

**Issue by sarnalisarkar** [VyomKulshrestha/Cortex-OS#382]
import streamlit as st
import pandas as pd
import numpy as np
from datetime import datetime, timedelta
import plotly.express as px
import plotly.graph_objects as go
from prophet import Prophet

# --- PAGE CONFIGURATION ---
st.set_page_config(
    page_title="Predictive Forecasting: Care Load & Placement Demand",
    layout="wide",
    initial_sidebar_state="expanded"
)

# --- SYNTHETIC DATA GENERATION (For Demonstration) ---
@st.cache_data
def generate_mock_data():
    np.random.seed(42)
    start_date = datetime(2024, 1, 1)
    date_list = [start_date + timedelta(days=x) for x in range(365 * 2)]
    
    # Base load with seasonality and random shocks (simulating border activity)
    base = 500
    trend = np.linspace(0, 200, len(date_list))
    seasonality = 150 * np.sin(np.array(range(len(date_list))) * (2 * np.pi / 365))
    shocks = np.random.choice([0, 300, 0, -150], size=len(date_list), p=[0.85, 0.05, 0.08, 0.02])
    noise = np.random.normal(0, 30, len(date_list))
    
    incoming_transfers = (base + trend + seasonality + shocks + noise).astype(int)
    incoming_transfers = np.clip(incoming_transfers, 100, None) # bound below
    
    # Discharge capacity roughly correlates but lags
    discharge_capacity = (incoming_transfers * 0.95 + np.random.normal(0, 20, len(date_list))).astype(int)
    
    df = pd.DataFrame({
        'Date': date_list,
        'Incoming_Transfers': incoming_transfers,
        'Discharge_Capacity': discharge_capacity,
        'Current_Care_Load': np.cumsum(incoming_transfers - discharge_capacity) + 1200
    })
    return df

df_data = generate_mock_data()

# --- SIDEBAR NAVIGATION ---
st.sidebar.title("Navigation")
page = st.sidebar.radio(
    "Go to:",
    [
        "Background & Context",
        "Problem Statement & Objectives",
        "Dataset Description",
        "Analytical Methodology",
        "Key Performance Indicators (KPIs)",
        "Forecasting Dashboard (App Requirements)"
    ]
)

# --- SECTION 1: BACKGROUND & CONTEXT ---
if page == "Background & Context":
    st.title("📘 Technical Documentation")
    st.header("Predictive Forecasting of Care Load & Placement Demand")
    st.caption("Detailed guide and project requirements for operational forecasting.")
    
    st.markdown("---")
    st.subheader("Background and Context")
    st.write(
        "The UAC Program operates in a **high-uncertainty environment**, where sudden changes "
        "in border activity, policy enforcement, or humanitarian crises can rapidly increase "
        "the number of children entering federal care."
    )
    st.write(
        "While descriptive analytics explain what has already happened, HHS decision-makers "
        "require forward-looking intelligence to answer:"
    )
    
    st.info("💡 **Key Decision Questions:**\n"
            "* How many children will be under HHS care in the coming days or weeks?\n"
            "* Will discharge capacity be sufficient to offset incoming transfers?")

# --- SECTION 2: PROBLEM STATEMENT & OBJECTIVES ---
elif page == "Problem Statement & Objectives":
    st.title("🎯 Project Focus Area")
    st.header("Problem Statement & Project Objectives")
    st.markdown("---")
    
    st.subheader("Problem Statement")
    st.error("Unpredictable fluctuations in incoming transfers create severe bottlenecks or underutilized capacity in care facilities, compromising the quality and speed of placement services.")
    
    st.subheader("Project Objectives")
    st.markdown("""
    * **Primary Objective:** Build a robust time-series forecasting pipeline to predict regional care loads 7, 14, and 30 days in advance.
    * **Secondary Objective:** Create an early-warning system that flags when predicted incoming transfers exceed historical discharge capacities.
    """)

# --- SECTION 3: DATASET DESCRIPTION ---
elif page == "Dataset Description":
    st.title("📊 Data Inventory")
    st.header("Dataset Description & Schema")
    st.markdown("---")
    
    st.write("Below is a preview of the historical operations dataset used for training the predictive models:")
    st.dataframe(df_data.head(10), use_container_width=True)
    
    st.subheader("Data Summary Statistics")
    st.dataframe(df_data.describe(), use_container_width=True)

# --- SECTION 4: ANALYTICAL METHODOLOGY ---
elif page == "Analytical Methodology":
    st.title("⚙️ Analytical Engineering")
    st.header("Analytical Methodology (Step-by-Step)")
    st.markdown("---")
    
    st.markdown("""
    ### 1. Data Ingestion & Preprocessing
    * Handling missing operational timestamps via forward-filling techniques.
    * Aggregating intake metrics to a uniform daily frequency.
    
    ### 2. Feature Engineering
    * Extraction of temporal attributes: Day of week, month, and holiday flags.
    * Generation of rolling window averages ($7$-day and $30$-day cycles) to capture short-term momentum.
    
    ### 3. Predictive Modeling
    * Utilizing **Prophet** models optimized for additive seasonal time-series patterns.
    * Implementing cross-validation using historical rolling windows.
    """)

# --- SECTION 5: KEY PERFORMANCE INDICATORS ---
elif page == "Key Performance Indicators (KPIs)":
    st.title("📈 Model Performance Framework")
    st.header("Key Performance Indicators (KPIs)")
    st.markdown("---")
    
    col1, col2, col3 = st.columns(3)
    with col1:
        st.metric(label="Target Mean Absolute Error (MAE)", value="< 15.2 children")
    with col2:
        st.metric(label="Target Mean Absolute Percentage Error (MAPE)", value="< 4.8%")
    with col3:
        st.metric(label="System Warning Reliability (Precision)", value="92%")
        
    st.markdown("""
    > **Note on Operational Metrics:** KPIs are tracked on an aggregate 14-day horizon window to ensure logistics teams have enough lead time to spin up temporary shelter accommodations.
    """)

# --- SECTION 6: STREAMLIT WEB APPLICATION REQUIREMENTS ---
elif page == "Forecasting Dashboard (App Requirements)":
    st.title("🔮 Operational Intelligence Engine")
    st.header("Predictive Forecasting Dashboard")
    st.markdown("---")
    
    # User Inputs for Interactive Forecasting
    st.sidebar.subheader("Forecast Parameters")
    horizon = st.sidebar.slider("Forecast Horizon (Days)", min_value=7, max_value=90, value=30)
    
    st.subheader(f"Simulating {horizon}-Day Operational Horizon")
    
    # Prepare data for Prophet
    prophet_df = df_data[['Date', 'Incoming_Transfers']].rename(columns={'Date': 'ds', 'Incoming_Transfers': 'y'})
    
    # Model Fit
    model = Prophet(yearly_seasonality=True, weekly_seasonality=True)
    model.fit(prophet_df)
    
    # Predict
    future = model.make_future_dataframe(periods=horizon)
    forecast = model.predict(future)
    
    # KPI Cards for Predicted State
    latest_load = df_data['Current_Care_Load'].iloc[-1]
    predicted_incoming_avg = forecast['yhat'].iloc[-horizon:].mean()
    historical_discharge_avg = df_data['Discharge_Capacity'].tail(30).mean()
    
    kpi1, kpi2, kpi3 = st.columns(3)
    with kpi1:
        st.metric(label="Current Care Load Capacity", value=f"{latest_load:,} Children")
    with kpi2:
        st.metric(label="Proj. Avg Daily Incoming (Next 30d)", value=f"{int(predicted_incoming_avg)} Units")
    with kpi3:
        # Check Alert status
        if predicted_incoming_avg > historical_discharge_avg:
            st.metric(label="System Capacity Alert Status", value="⚠️ HIGH RISK", delta="Demand Exceeds Discharge")
        else:
            st.metric(label="System Capacity Alert Status", value="✅ STABLE", delta="Within Capacity Limits")
            
    # Plotly Visualizations
    st.subheader("Predictive Demand vs System Limits")
    
    fig = go.Figure()
    # Historical
    fig.add_trace(go.Scatter(x=df_data['Date'].tail(90), y=df_data['Incoming_Transfers'].tail(90), name="Historical Intake", line=dict(color='gray')))
    # Forecast
    fig.add_trace(go.Scatter(x=forecast['ds'].iloc[-horizon:], y=forecast['yhat'].iloc[-horizon:], name="Forecasted Intake", line=dict(color='blue', width=3)))
    # Confidence Intervals
    fig.add_trace(go.Scatter(
        x=pd.concat([forecast['ds'].iloc[-horizon:], forecast['ds'].iloc[-horizon:][::-1]]),
        y=pd.concat([forecast['yhat_upper'].iloc[-horizon:], forecast['yhat_lower'].iloc[-horizon:][::-1]]),
        fill='toself',
        fillcolor='rgba(0,0,255,0.1)',
        line=dict(color='rgba(255,255,255,0)'),
        hoverinfo="skip",
        showlegend=True,
        name="Confidence Interval"
    ))
    
    fig.update_layout(title="Incoming Transfers Forecast Analysis", xaxis_title="Timeline", yaxis_title="Volume Counts", template="plotly_white")
    st.plotly_chart(fig, use_container_width=True)

**Comment by kaushtubhtiwarigit** [comment]
@VyomKulshrestha  kindly assign me this issue

**Comment by Pro-Joshi-grammer** [comment]
@sarnalisarkar  Could you clarify the issue here with words?

**Comment by khushikr7979** [comment]
Hi @sarnalisarkar,

Could you please clarify the issue requirements?

The description currently contains the Streamlit dashboard code, but I am unable to identify the specific bug, enhancement, or expected task from the provided information.

Could you provide more details on what needs to be implemented or fixed?

Thank you!
